src/formatDate.js

```diff
@@ -1,14 +1,66 @@
 'use strict';
 
-/**
- * @param {string} date
- * @param {string[]} fromFormat
- * @param {string[]} toFormat
- *
- * @returns {string}
- */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateComponents = date.split(fromFormat[fromFormat.length - 1]);
+  const formatedDate = [];
+  let day = '';
+  let month = '';
+  let year = '';
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    switch (fromFormat[i]) {
+      case 'DD':
+        day = dateComponents[i];
+        break;
+
+      case 'MM':
+        month = dateComponents[i];
+        break;
+
+      case 'YY':
+        year = dateComponents[i];
+        break;
+
+      case 'YYYY':
+        year = dateComponents[i];
+        break;
```

Comment:
```suggestion
      case 'YY':
      case 'YYYY':
        year = dateComponents[i];
        break;
```
- You can combine cases if they do the same
- And you still must have default case to handle unexpected input